src/axom/primal/tests/primal_bezier_intersect.cpp

```diff
@@ -0,0 +1,364 @@
+  BezierCurveType curve2(data2, order);
+
+  // Note: same intersection params for curve and line
+  std::vector<CoordType> exp_intersections = { 0.17267316464601146,
+                                               0.5,
+                                               0.827326835353989};
+
+  for(int exp=1 ; exp<=16 ; ++exp)
+  {
+
+    const double eps = std::pow(10, -exp);
+    const double eps_test = std::pow(10, -exp+1);
+    SLIC_INFO("Testing w/ eps = " << eps );
+    std::stringstream sstr;
+    sstr<<"cubic eps study " << eps;
+    SCOPED_TRACE(sstr.str());
+
+    checkIntersections(curve1, curve2,
+                       exp_intersections, exp_intersections,
+                       eps, eps_test);
+  }
+
+}
+
+
+TEST( primal_bezier_inter, cubic_bezier_nine_intersections )
+{
+  static const int DIM =2;
+  using CoordType = double;
+  using PointType = primal::Point< CoordType, DIM >;
+  using BezierCurveType = primal::BezierCurve< CoordType, DIM >;
+
+  SLIC_INFO("primal: testing bezier intersection");
+  SCOPED_TRACE("cubic bezier -- 9 intersections");
+
+  // A configuration of two cubic bezier curves that intersect at nine points
+  const int order = 3;
+  PointType data1[order+1] =  { PointType::make_point(100,  90),
+                                PointType::make_point(125, 260),
+                                PointType::make_point(125,   0),
+                                PointType::make_point(140, 145)};
+
+  BezierCurveType curve1(data1, order);
+
+  PointType data2[order+1] =  { PointType::make_point( 75, 110),
+                                PointType::make_point(265, 120),
+                                PointType::make_point(  0, 130),
+                                PointType::make_point(145, 135)};
+  BezierCurveType curve2(data2, order);
+
+  const double eps = 1E-16;
+  const double eps_test = 1E-10;
+
+  std::vector<CoordType> exp_s = { 0.04832125363145223,
+                                   0.09691296096235966,
+                                   0.1149546049907844,
+                                   0.4525395443158645,
+                                   0.5130753010911715,
+                                   0.5874279469012619,
+                                   0.9096761427522992,
+                                   0.937347647827529,
+                                   0.9747492689591496};
+
+  std::vector<CoordType> exp_t = { 0.05756422605799361,
+                                   0.1237935342287382,
+                                   0.1676610724548464,
+                                   0.4229476589108138,
+                                   0.518580927287544,
+                                   0.6475905573955213,
+                                   0.872202573839462,
+                                   0.9370948395920837,
+                                   0.9853581324900483};
+
+  checkIntersections(curve1, curve2,
+                     exp_s, exp_t,
+                     eps, eps_test);
+}
+
```

Comment:
Consider adding
1. Intersection tests between Bezier curves of different orders
2. Test non-intersecting curves
3. Test curves embedded in 3-space
4. Tangent intersections; identical (sections of) curve (if the algorithm will handle this)

Reply:
I think commit https://github.com/LLNL/axom/pull/59/commits/1d8a83ddc3468fd8d5054f1894db9d4efc741e42 addresses 1. and 2. and 3. I've also created issue #71 , which will require more thought to deal with 4.

Reply:
Thanks!  Looks good.